### PR TITLE
#341 deletes all jobs associated to a company when the company is unm…

### PR DESCRIPTION
**Pull request by fullKTRL**
Deletes all jobs associated to a company when the company is unmarked.

### CODE REVIEW

File: synapsis/src/main/java/com/soen/synapsis/appuser/AppUserService.java [hunk 1]
```diff
@@ -329,6 +334,12 @@ public void markCompanyAsNonVerified(AppUser appUser) {
                 appUserRepository.save(recruiter);
             }
         }
+
+        List<Job> jobsToDelete = jobRepository.findJobsByCompanyEquals(appUser);
+        for (Job jobToDelete : jobsToDelete) {
+            jobRepository.delete(jobToDelete);
+        }
```
Comment (Ghaith-Chrit) [inline synapsis/src/main/java/com/soen/synapsis/appuser/AppUserService.java:341]: You can replace the for loop with just `jobRepository.deleteAll(jobsToDelete);`

**Review by Ghaith-Chrit — APPROVED** [review]
LGTM

File: synapsis/src/main/java/com/soen/synapsis/appuser/AppUserService.java (hunk 1 above)
Comment (fullKTRL) [reply synapsis/src/main/java/com/soen/synapsis/appuser/AppUserService.java:341]: I didn't implement that change, I'm not sure how it's supposed to be.  I'll replace the loop.

**Comment by Chr728** [comment]
> LGTM. Btw I think you should change the role of the recruiters of the company to `CANDIDATE` not just set their verification status to false since they can still create new jobs even if the company is unverified unless I am missing something.

@Ghaith-Chrit I think it is okay if we leave it as recruiters because recruiters initially at the start before they get marked for the first time they have role type recruiter not candidates, so when they get unmarked they are returning back to their original state which is a recruiter, not a candidate ( although they have same access to services similar to candidates )

**Pull request merged**